File: programmers/코딩테스트 입문 with Java/순서쌍의개수.java
package 코딩테스트입문;

/**
 * https://school.programmers.co.kr/learn/courses/30/lessons/120836
 */
public class 순서쌍의개수 {
    public static void main(String[] args) {
//        System.out.println(solution(20));
//        System.out.println(solution(100));
//        System.out.println(solution(30));
        System.out.println(solution(1));
    }

    public static int solution(int n) {
        int answer = 0;
        if (n == 1) return 1;
        for (int i = 1; i <= n / 2; i++) {
            if (n % i == 0) answer++;
            if (n / 2 == i) answer++;
        }

        return answer;
    }
}

File: programmers/코딩테스트 입문 with Python/순서쌍의 개수.py
def solution(n):
    answer = 0

    for i in range(1, n + 1):
        a, b = divmod(n, i)
        if b == 0:
            answer += 1
        if i >= n // 2:
            if i == n // 2:
                answer += 1
            break
    return answer


print(solution(20))
